pkg/ssh/types/connections.go

```diff
@@ -0,0 +1,212 @@
+package types
```

Comment:
Do we need a dedicated types package, or can we merge this into `pkg/ssh`?

Note that `pkg/ssh/foo` can import `pkg/ssh`.

Reply:
go is screaming `import cycle not allowed` when I do this

Reply:
this is because ssh/runtime.go uses `package native` and `package golang` to create the proper `ConnectionEngine`. I think this is an ok tradeoff for simplicity.

Reply:
so @vrothberg in podman, calling all of these functions one would need to specify if they want to use golang or native ssh? or would this be a cmd level podman flag? I am a fan of this option as I think it adds more flexibility.

I could see a root level podman flag being applicable here.... WDYT @rhatdan @jwhonce ?

Reply:
@cdoern, my comment should not change the way this decision is made.

I image a global flag and an option in containers.conf.

Reply:
@vrothberg mocking it up now and I think you are right, it is much much simpler. The podman implementation can stay the same as long as we have a way for users to specify what they want. 

this is still an "interface" in that the top level functions are all that can be called (Dial, Scp, and Create), but the choice is passed from podman not generated from common

Reply:
Thanks a ton for considering this option, @cdoern!